### ISSUE TITLE
Preventing validators to run on `draft` PRs.

### ISSUE THREAD

**Issue by hemeroc** [mergeability/mergeable#565]
If I understand this PR https://github.com/mergeability/mergeable/pull/515 correct the new payload filter enables a user to create a filter for any payload element.

**Expectation:**

> Therefore my expectation would be that I can use that filter to run specific validators not on `draft` PRs.
> As far as I have seen that does not work due to the fact that the filter does not support boolean/integer checks.

**Reason:**

>I would need this feature mainly to prevent `stale` checks from running on `draft` PRs.

Please let me know if my assumptions are correct or if I am thinking in a wrong direction.
Other options would be:
- adding a validator for `darft` PRs
- making this an option of the `stale` validator

Also let me know if I can support you adding this possibility.

**Comment by shine2lay** [comment]
@hemeroc I think you are right, the only thing preventing this use case is the lack of checks against `boolean` in `payload` filter.

I think you'll need to add the `boolean` option to this line https://github.com/mergeability/mergeable/blob/master/lib/filters/payload.js#L7 
and add something similar to this option processor https://github.com/mergeability/mergeable/blob/master/lib/validators/options_processor/options/no_empty.js 
within the `lib/filters/options` folder

I won't be able to get to this anytime soon, if you do create a PR however, i'll make sure it gets merged ASAP

**Comment by hemeroc** [comment]
For ease of implementation I went for implementing an option on the the stale validator called `ignore_drafts`.
Similat to the project and milestone options.

**Comment by shine2lay** [comment]
@bbugh yes, I was recommending something like that, to be consistent with current implementation. do you have a recommendation?

**Comment by bbugh** [comment]
No better recommendations. :) Now that I'm clear on what you were proposing, I will submit a PR for this.

**Comment by shine2lay** [comment]
@bbugh awesome! 
What do you think about using `on` instead of `enabled`? I think `enable` is not as intuitive if you just look at the config. 

- do: payload
        pull_request:
          draft:
            boolean:
              # don't run on `draft: true` (since this is a filter, only if the condition is true, the validation should would run)
              on: false

**Comment by bbugh** [comment]
I was playing with `match` like some of the other options, since we want to validate that it's an exact value. There's also `is`. I think the equality synonym is useful. Like:

```
      - do: payload
        pull_request:
          draft:
            boolean:
              # like current options
              match: false
			  # or
              is: false
```

**Comment by shine2lay** [comment]
@bbugh i think you are right, `match` would better IMO too. it also keeps the consistency with the rest of the config

**Comment by bbugh** [comment]
Draft PR! https://github.com/mergeability/mergeable/pull/583
